Moto 5.0.12 breaks amazon provider's RDS tests

### Body

The **just released** moto `5.0.12` breaks tests for AWS RDS 

https://github.com/apache/airflow/actions/runs/10230423335/job/28305276935#step:7:919

The way to reproduce it:

* Enter breeze `breeze` (you get latest constraints dependencies there)
* Run the test `pytest tests/providers/amazon/aws/sensors/test_rds.py::TestRdsExportTaskExistenceSensor::test_export_task_poke_false`
* It succeeds
* Upgrade moto to latest version `pip install --upgrade moto`:

```
Installing collected packages: moto
  Attempting uninstall: moto
    Found existing installation: moto 5.0.11
    Uninstalling moto-5.0.11:
      Successfully uninstalled moto-5.0.11
Successfully installed moto-5.0.12
```
* run the same test again

Fails with 

```
  raise error_class(parsed_response, operation_name)
E           botocore.errorfactory.ExportTaskNotFoundFault: An error occurred (ExportTaskNotFound) when calling the DescribeExportTasks operation: Cannot cancel export task because a task with the identifier my-db-instance-snap-export is not exist.
```


### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

--- a/airflow/providers/amazon/aws/hooks/rds.py
+++ b/airflow/providers/amazon/aws/hooks/rds.py
@@ -155,7 +155,7 @@ class RdsHook(AwsGenericHook["RDSClient"]):
         try:
             response = self.conn.describe_export_tasks(ExportTaskIdentifier=export_task_id)
         except self.conn.exceptions.ClientError as e:
-            if e.response["Error"]["Code"] == "ExportTaskNotFoundFault":
+            if e.response["Error"]["Code"] in ("ExportTaskNotFound", "ExportTaskNotFoundFault"):
                 raise AirflowNotFoundException(e)
             raise e
         return response["ExportTasks"][0]["Status"].lower()
@@ -196,7 +196,7 @@ class RdsHook(AwsGenericHook["RDSClient"]):
         try:
             response = self.conn.describe_event_subscriptions(SubscriptionName=subscription_name)
         except self.conn.exceptions.ClientError as e:
-            if e.response["Error"]["Code"] == "SubscriptionNotFoundFault":
+            if e.response["Error"]["Code"] in ("SubscriptionNotFoundFault", "SubscriptionNotFound"):
                 raise AirflowNotFoundException(e)
             raise e
         return response["EventSubscriptionsList"][0]["Status"].lower()

--- a/airflow/providers/amazon/provider.yaml
+++ b/airflow/providers/amazon/provider.yaml
@@ -139,9 +139,7 @@ additional-extras:
 devel-dependencies:
   - aiobotocore>=2.13.0
   - aws_xray_sdk>=2.12.0
-  # The latest 5.0.12 moto breaks some of our tests https://github.com/apache/airflow/issues/41243
-  # Until we fix it, we limit moto to <5.0.12
-  - moto[cloudformation,glue]>=5.0.0,<5.0.12
+  - moto[cloudformation,glue]>=5.0.0
   - mypy-boto3-appflow>=1.34.0
   - mypy-boto3-rds>=1.34.90
   - mypy-boto3-redshift-data>=1.34.0

--- a/generated/provider_dependencies.json
+++ b/generated/provider_dependencies.json
@@ -46,7 +46,7 @@
     "devel-deps": [
       "aiobotocore>=2.13.0",
       "aws_xray_sdk>=2.12.0",
-      "moto[cloudformation,glue]>=5.0.0,<5.0.12",
+      "moto[cloudformation,glue]>=5.0.0",
       "mypy-boto3-appflow>=1.34.0",
       "mypy-boto3-rds>=1.34.90",
       "mypy-boto3-redshift-data>=1.34.0",

